### ISSUE TITLE
Link `SolidusStripe::Customer` to reusable payment sources

### ISSUE THREAD

**Issue by elia** [solidusio-contrib/solidus_stripe#164]
## What

Associate it to the stripe customer id that was created for the order's user whenever a payment source is created for a reusable profile.

See also https://github.com/solidusio/solidus_braintree/blob/c2c6452bb54244afce6af4f27d81b081da10bff6/app/models/solidus_braintree/gateway.rb#L227-L249.

### Dependencies

- Blocked by #153
- Blocked by #163 

## Why

Payments should be tied to a single customer on stripe as they're coming from a single user on solidus, this will make handling issues and investigations much easier (e.g. from customer support team members).

**Comment by loicginoux** [comment]
This might be for another ticket, let me know but when having a customer created and linked to a payment intent, from my understanding of Stripe, iOS and Android clients will also need an ephemeral key as described in the [Stripe documentation ](https://stripe.com/docs/payments/accept-a-payment?platform=ios&ui=payment-sheet#add-server-endpoint)
Something like:
```ruby
ephemeral_key = Stripe::EphemeralKey.create({
  customer: customer['id'],
}, { stripe_version: Stripe.api_version})
```
and then the PaymentIntentController#create will respond with all the info need o that any kind of api client can handle it

```ruby
render json: { 
  client_secret: intent.client_secret,
  ephemeral_key: ephemeral_key['secret'],
  customer: customer['id'],
}
```

**Comment by elia** [comment]
For serving iOS clients it's probably best to add a dedicated controller under Spree::Api and return what's needed there.

In the latest branch (which we'll hopefully merge soon) the secret is not fetched via Ajax anymore and it's instead being provided by to the payment view directly.

**Comment by elia** [comment]
@loicginoux I extracted your comment about iOS/android to #207, so we don't lose track

**Comment by elia** [comment]
Fixed in #172
